File: models/Contract.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Contract = sequelize.define('Contract', {
    serviceType: {
      type: DataTypes.ENUM('agency', 'dispatch', 'outsourcing'),
      allowNull: false
    },
    startDate: {
      type: DataTypes.DATE,
      allowNull: false
    },
    endDate: DataTypes.DATE,
    terms: DataTypes.TEXT,
    status: {
      type: DataTypes.ENUM('active', 'completed', 'terminated'),
      defaultValue: 'active'
    }
  });

  return Contract;
};

File: models/contract.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Contract struct {
	gorm.Model
	ClientID    uint      `json:"clientId" binding:"required"`
	ServiceType string    `json:"serviceType" binding:"required"`
	StartDate   time.Time `json:"startDate" binding:"required"`
	EndDate     time.Time `json:"endDate"`
	Terms       string    `json:"terms"`
	Status      string    `json:"status" binding:"required"`
}